sqlx-core/src/postgres/options/mod.rs

```diff
@@ -319,6 +323,34 @@ impl PgConnectOptions {
         self
     }
 
+    /// Set additional startup options for the connection as a list of key-value pairs.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use sqlx_core::postgres::PgConnectOptions;
+    /// let options = PgConnectOptions::new()
+    ///     .options([("geqo", "off"), ("statement_timeout", "5min")]);
+    /// ```
+    pub fn options<K, V, I>(mut self, options: I) -> Self
+    where
+        K: Display,
+        V: Display,
+        I: IntoIterator<Item = (K, V)>,
+    {
+        let mut options_str = String::new();
+        for (k, v) in options {
+            options_str += &format!("-c {}={}", k, v);
```

Comment:
This appears to have a bug where if multiple options are provided the `-c` is tacked onto the previous one. This can be worked around by adding the space manually.

```rust
		.options([
			("default_transaction_isolation", "serializable "),
			("idle_in_transaction_session_timeout", "5min "),
			("statement_timeout", "2s"),
		]);
```

Or by calling this multiple times.

Reply:
Bug filed: https://github.com/launchbadge/sqlx/issues/1730